content-retrievers/langchain4j-community-lucene/src/main/java/dev/langchain4j/rag/content/retriever/lucene/LuceneContentRetriever.java

```diff
@@ -0,0 +1,360 @@
+         * Sets the name of the token count field.
+         *
+         * @param tokenCountFieldName Token count field name
+         * @return Builder
+         */
+        public LuceneContentRetrieverBuilder tokenCountFieldName(final String tokenCountFieldName) {
+            if (tokenCountFieldName == null || tokenCountFieldName.isBlank()) {
+                this.tokenCountFieldName = LuceneIndexer.TOKEN_COUNT_FIELD_NAME;
+            }
+            this.tokenCountFieldName = tokenCountFieldName;
+
+            return this;
+        }
+
+        /**
+         * Returns only a certain number of documents.
+         *
+         * @param topNMatches Number of documents to return
+         * @return Builder
+         */
+        public LuceneContentRetrieverBuilder topNMatches(final int topNMatches) {
+            if (topNMatches >= 0) {
+                this.topNMatches = topNMatches;
+            }
+            return this;
+        }
+    }
+
+    private static final Logger LOGGER = Logger.getLogger(LuceneContentRetriever.class.getCanonicalName());
+
+    /**
+     * Instantiate a builder for `LuceneContentRetriever`.
+     *
+     * @return Builder for `LuceneContentRetriever`
+     */
+    public static LuceneContentRetrieverBuilder builder() {
+        return new LuceneContentRetrieverBuilder();
+    }
+
+    private final Directory directory;
+    private final boolean onlyMatches;
+    private final int topNMatches;
+    private final int maxTokenCount;
+    private final String contentFieldName;
+    private final String tokenCountFieldName;
+
+    /**
+     * Initialize all fields, and do one more round of validation (even though the builder has
+     * validated the fields).
+     *
+     * @param directory Lucene directory
+     * @param onlyMatches Whether to only consider matching documents
+     * @param topNMatches Return only the first n matches
+     * @param maxTokenCount Return until a maximum token count
+     * @param contentFieldName Name of the Lucene field with the text
+     * @param tokenCountFieldName Name of the Lucene field with token counts
+     */
+    private LuceneContentRetriever(
+            final Directory directory,
+            final boolean onlyMatches,
+            final int topNMatches,
+            final int maxTokenCount,
+            final String contentFieldName,
+            final String tokenCountFieldName) {
+        this.directory = requireNonNull(directory, "No directory provided");
+        this.onlyMatches = onlyMatches;
+        this.topNMatches = Math.max(0, topNMatches);
+        this.maxTokenCount = Math.max(0, maxTokenCount);
+        this.contentFieldName = requireNonNull(contentFieldName, "No content field name provided");
+        if (this.contentFieldName.isBlank()) {
+            throw new IllegalArgumentException("Content field name cannot be blank");
+        }
+        this.tokenCountFieldName = requireNonNull(tokenCountFieldName, "No token count field name provided");
```

Comment:
```suggestion
       this.tokenCountFieldName = ensureNotBlank(tokenCountFieldName, "tokenCountFieldName");
```

Just using `ensureNotBlank` to check :)